File: algorithms/289.Game_of_Life/Game_of_Life.py
class Solution(object):
    def gameOfLife(self, board):
    # update the board based on number
    # 0: dead to dead
    # 1: live to live
    # 2: live to dead (1 -> 0)
    # 3: dead to live (0 -> 1)
            drx = [-1, 0,  1, -1, 1, -1, 0, 1]
            dry = [-1, -1, -1, 0, 0,  1, 1, 1]
            n = len(board)
            m = len(board[0])
        
            for i in xrange(n):
                for j in xrange(m):
                    now = board[i][j]
                    nigh_live = 0
                    for k in xrange(8):
                        newx = j+drx[k]
                        newy = i+dry[k]
                        if newx<0 or newx>=m or newy<0 or newy>=n:
                            continue
                        if board[newy][newx] == 1 or board[newy][newx] == 2:
                            nigh_live += 1
        
                    if now == 1 and (nigh_live < 2 or nigh_live > 3):
                        board[i][j] = 2
                    elif now == 0 and nigh_live == 3:
                        board[i][j] = 3
        
            for i in xrange(n):
                for j in xrange(m):
                    board[i][j] %= 2

File: algorithms/289.Game_of_Life/Game_of_Life.java
<<<<<<< HEAD
public class Solution {
int[][] dir ={{1,-1},{1,0},{1,1},{0,-1},{0,1},{-1,-1},{-1,0},{-1,1}};
public void gameOfLife(int[][] board) {
    for(int i=0;i<board.length;i++){
        for(int j=0;j<board[0].length;j++){
            int live=0;
            for(int[] d:dir){
                if(d[0]+i<0 || d[0]+i>=board.length || d[1]+j<0 || d[1]+j>=board[0].length) continue;
                if(board[d[0]+i][d[1]+j]==1 || board[d[0]+i][d[1]+j]==2) live++;
            }
            if(board[i][j]==0 && live==3) board[i][j]=3;
            if(board[i][j]==1 && (live<2 || live>3)) board[i][j]=2;
        }
    }
    for(int i=0;i<board.length;i++){
        for(int j=0;j<board[0].length;j++){
            board[i][j] %=2;
        }
    }
=======
public class Solution {
int[][] dir ={{1,-1},{1,0},{1,1},{0,-1},{0,1},{-1,-1},{-1,0},{-1,1}};
public void gameOfLife(int[][] board) {
    for(int i=0;i<board.length;i++){
        for(int j=0;j<board[0].length;j++){
            int live=0;
            for(int[] d:dir){
                if(d[0]+i<0 || d[0]+i>=board.length || d[1]+j<0 || d[1]+j>=board[0].length) continue;
                if(board[d[0]+i][d[1]+j]==1 || board[d[0]+i][d[1]+j]==2) live++;
            }
            if(board[i][j]==0 && live==3) board[i][j]=3;
            if(board[i][j]==1 && (live<2 || live>3)) board[i][j]=2;
        }
    }
    for(int i=0;i<board.length;i++){
        for(int j=0;j<board[0].length;j++){
            board[i][j] %=2;
        }
    }
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
}